src/frame-writer.cpp

```diff
@@ -447,15 +476,11 @@ void FrameWriter::add_frame(const uint8_t* pixels, int64_t usec, bool y_invert)
     pkt.data = NULL;
     pkt.size = 0;
 
-    int got_output;
-    avcodec_encode_video2(videoCodecCtx, &pkt, *output_frame, &got_output);
+    encode(videoCodecCtx, *output_frame, &pkt);
 
     /* Restore frame buffer, so that it can be properly freed in the end */
     if (saved_buf0)
```

Comment:
IIRC this should happen before `finish_frame` as was in the original code, otherwise we may be leaking memory.

Reply:
`finish_frame` is called in encode() in a while loop, after avcodec_send_frame(). I think if you look how to use avcodec_send_frame() API, you will see this is correct.

Reply:
Just looked into it again, seems you're right.